src/main/java/org/killbill/billing/client/KillBillClient.java

```diff
@@ -1913,14 +1946,25 @@ private String getResourceFile(final String uri, final String contentType) throw
 
     private <ReturnType> ReturnType uploadFile(final String fileToUpload, final String uri, final String contentType, final String createdBy, final String reason, final String comment, final Class<ReturnType> followUpClass) throws KillBillClientException {
         Preconditions.checkNotNull(fileToUpload, "fileToUpload cannot be null");
+        final File catalogFile = new File(fileToUpload);
+        Preconditions.checkArgument(catalogFile.exists() && catalogFile.isFile() && catalogFile.canRead(), "file to upload needs to be a valid file");
+        try {
+            final InputStream fileStream = Files.asByteSource(catalogFile).openStream();
```

Comment:
Shouldn't the `fileStream` be closed in a `finally` block here (I assume the `uploadFile` method below doesn't close the `fileToUpload` on purpose)?